### ISSUE TITLE
(Docs - Concepts) Hyperparameters

### ISSUE THREAD

**Issue by guenthermi** [jina-ai/finetuner#729]
- explain the batch size and number of epochs
- explain limitations on setting the batch size
- explain the automatic configuration of the batch size and handling of out-of-memory errors because of too high batch sizes
- maybe shortly reference optimizers and loss function sections with the hyperparameters discussed there, e.g.g since optimizer parameters are also considered as hyperparameters
